This patch adds a 32-bit i_version_hi field to ext4_inode, which can be used for 64-bit inode versions. This field will store the higher 32 bits of the version, while Jean Noel's patch has added support to store the lower 32-bits in osd1.linux1.l_i_version.

Signed-off-by: Andreas Dilger <adilger@clusterfs.com>
Signed-off-by: Kalpak Shah <kalpak@clusterfs.com>
---
Index: linux-2.6.21-rc7/include/linux/ext4_fs.h
===================================================================
--- linux-2.6.21-rc7.orig/include/linux/ext4_fs.h	2007-04-17 14:26:25.000000000 -0700
+++ linux-2.6.21-rc7/include/linux/ext4_fs.h	2007-04-17 17:54:16.000000000 -0700
@@ -349,6 +349,7 @@
 	__le32  i_atime_extra;  /* extra Access time      (nsec << 2 | epoch) */
 	__le32  i_crtime;       /* File Creation time */
 	__le32  i_crtime_extra; /* extra File Creation time (nsec << 2 | epoch) */
+	__le32   i_version_hi;   /* high 32 bits for 64-bit version */
 };
 
 #define i_size_high	i_dir_acl
